nixos/modules/virtualisation/docker.nix

```diff
@@ -40,13 +40,25 @@ in
       };
     storageDriver =
       mkOption {
-        type = types.enum ["aufs" "btrfs" "devicemapper" "overlay" "zfs"];
-        default = "devicemapper";
+        type = types.nullOr (types.enum ["aufs" "btrfs" "devicemapper" "overlay" "zfs"]);
```

Comment:
Note that there is a `overlay2` driver with 1.12. The only requirement for the driver is a kernel > 4.0 if I remember correctly.